tests/AppBundle/Controller/AdherentControllerTest.php

```diff
@@ -251,6 +252,54 @@ public function testAdherentChangePassword()
         $this->authenticateAsAdherent($this->client, 'carl999@example.fr', 'heaneaheah');
     }
 
+    public function testAdherentSetEmailNotifications()
+    {
+        $adherent = $this->getAdherentRepository()->findByEmail('carl999@example.fr');
+
+        $this->assertFalse($adherent->hasSubscribedLocalHostEmails());
+        $this->assertFalse($adherent->hasSubscribedReferentsEmails());
+        $this->assertFalse($adherent->hasSubscribedMainEmails());
+
+        $this->authenticateAsAdherent($this->client, 'carl999@example.fr', 'secret!12345');
+
+        $crawler = $this->client->request(Request::METHOD_GET, '/espace-adherent/mon-profil/preferences-des-emails');
+        $subscriptions = $crawler->filter('input[name="adherent_email_subscription[email_subscriptions][]"]');
+
+        $this->assertCount(3, $subscriptions);
+
+        // Submit the emails subscription form with invalid data
+        $crawler = $this->client->request(Request::METHOD_POST, '/espace-adherent/mon-profil/preferences-des-emails', [
+            'adherent_email_subscription' => [
+                'email_subscriptions' => ['heah'],
+                '_token' => $crawler->filter('input[name="adherent_email_subscription[_token]"]')->attr('value'),
```

Comment:
You should document why you're using $client->request() instead of $client->submit($form) in this special case.